Add simde

jcarpent opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

jcarpent commented:
@conda-forge/help-c-cpp, ready for review!

saraedum commented on recipes/simde/meta.yaml:
```diff
@@ -0,0 +1,42 @@
+  url: https://github.com/simd-everywhere/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 366d5e9a342c30f1e40d1234656fb49af5ee35590aaf53b3c79b2afb906ed4c8
+
+build:
```
If I understand correctly, this is a header-only library? Is there any dependency on the compiler that was used to build this package? If not, you could add something like the following

```
  ignore_run_exports_from:
    # Since this package does not ship any compiled code, we do not need any
    # compatibility with the compiler used at runtime.
    - {{ compiler('cxx') }}
```

jcarpent replied:
Done.

saraedum commented on recipes/simde/meta.yaml:
```diff
@@ -0,0 +1,42 @@
+
+build:
+  number: 0
+  skip: true  # [win]
```
If this is header-only, why is Windows not supported?

jcarpent replied:
Done.

saraedum commented on recipes/simde/meta.yaml:
```diff
@@ -0,0 +1,42 @@
+build:
+  number: 0
+  skip: true  # [win]
+  run_exports:
```
Why do we need a `run_exports` if this package consists only of header files?

jcarpent replied:
Done.

jcarpent commented:
Thanks @saraedum for the complete review. I've addressed all the comments you made.
https://github.com/orgs/conda-forge/teams/help-c-cpp, ready for review!

jcarpent commented:
https://github.com/orgs/conda-forge/teams/help-c-cpp, ready for review!

jcarpent commented:
@conda-forge/help-c-cpp, I've handled all the comments. Could you make a second pass?
Thanks in advanced

jcarpent commented:
@conda-forge/help-c-cpp, ready for review!

saraedum commented on recipes/simde/meta.yaml:
```diff
@@ -0,0 +1,43 @@
+    - pkg-config
+  commands:
+    - pkg-config --modversion {{ name }}
+    - test -f $PREFIX/include/simde/simde-common.h  # [not win]
```
```suggestion
    - test -f $PREFIX/include/simde/simde-common.h  # [not win]
    - if not exist %LIBRARY_INC%\simde\simde-common.h exit 1  # [win]
```

jcarpent commented:
@conda-forge/help-c-cpp, ready for review!

jcarpent commented:
Perfect. Thanks a lot @saraedum for all your feedback.
